src/components/ec-datepicker/ec-datepicker.spec.js

```diff
@@ -0,0 +1,387 @@
+    }
+  });
+
+  function mountDatepicker(props, mountOpts) {
+    const inputWrapper = mount(EcDatepicker, {
+      propsData: { ...props },
+      ...mountOpts,
+    });
+
+    const calendarWrapper = createWrapper(document.body).findByDataTest('ec-datepicker__calendar');
+
+    return { inputWrapper, calendarWrapper };
+  }
+
+  function mountDatepickerAsTemplate(template, props, wrapperComponentOpts, mountOpts) {
+    const localVue = createLocalVue();
+
+    const Component = localVue.extend({
+      components: { EcDatepicker },
+      template,
+      ...wrapperComponentOpts,
+    });
+
+    const inputWrapper = mount(Component, {
+      propsData: { ...props },
+      ...mountOpts,
+    });
+
+    const calendarWrapper = createWrapper(document.body).findByDataTest('ec-datepicker__calendar');
+
+    return { inputWrapper, calendarWrapper };
+  }
+
+  it('should render properly', () => {
+    const { inputWrapper } = mountDatepicker();
+
+    expect(inputWrapper.element).toMatchSnapshot();
+  });
+
+  describe(':props', () => {
+    it('should render with a label', () => {
+      const { inputWrapper } = mountDatepicker({
+        label: 'Datepicker label',
+      });
+
+      expect(inputWrapper.element).toMatchSnapshot();
+    });
+
+    it('should render with a note', () => {
+      const { inputWrapper } = mountDatepicker({
+        note: 'Datepicker note',
+      });
+
+      expect(inputWrapper.element).toMatchSnapshot();
+    });
+
+    it('should render with a bottom note', () => {
+      const { inputWrapper } = mountDatepicker({
+        bottomNote: 'Datepicker bottom note',
+      });
+
+      expect(inputWrapper.element).toMatchSnapshot();
+    });
+
+    it('should render with a bottom note in a warning state', () => {
+      const { inputWrapper } = mountDatepicker({
+        bottomNote: 'Datepicker bottom note',
+        isWarning: true,
+      });
+
+      expect(inputWrapper.element).toMatchSnapshot();
+    });
+
+    it('should render with a placeholder', () => {
+      const { inputWrapper } = mountDatepicker({
+        placeholder: 'Datepicker placeholder',
+      });
+
+      expect(inputWrapper.element).toMatchSnapshot();
+    });
+
+    it('should render with an error message', () => {
+      const { inputWrapper } = mountDatepicker({
+        errorMessage: 'A random error message',
+      });
+
+      expect(inputWrapper.element).toMatchSnapshot();
+    });
+
+    it('should render as disabled', () => {
+      const { inputWrapper } = mountDatepicker({
+        isDisabled: true,
+      });
+
+      expect(inputWrapper.element).toMatchSnapshot();
+    });
+
+    it('should have disabled a specific date', () => {
+      const { calendarWrapper } = mountDatepicker({
+        disabledDates: {
+          '2022-02-22': 'Bank holiday',
+          '2022-02-23': '',
+        },
+      });
+
+      expect(calendarWrapper.findByDataTest('ec-datepicker__calendar-day--2022-02-22').classes('flatpickr-disabled')).toBeTruthy();
```

Comment:
`classes()` return boolean so we can safely test `.toBe(true)` instead.

Reply:
Sure will amend. @mcibique Just out of curiousity when we should choose one over the other?